tests/src/Functional/ReplicationSettingsTest.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Drupal\Tests\workspace\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Drupal\multiversion\Entity\Workspace;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+use Drupal\simpletest\BlockCreationTrait;
+use Drupal\workspace\Entity\WorkspacePointer;
+
+/**
+ * Tests replication settings with the internal replicator.
+ *
+ * @group workspace
+ */
+class ReplicationSettings extends BrowserTestBase {
+
+  use WorkspaceTestUtilities;
+
+  use BlockCreationTrait {
+    placeBlock as drupalPlaceBlock;
+  }
+
+  public static $modules = [
+    'system',
+    'node',
+    'user',
+    'block',
+    'workspace',
+    'multiversion',
+    'taxonomy',
+    'entity_reference',
+    'field',
+    'field_ui',
+    'menu_link_content',
+    'menu_ui',
+    'replication',
+  ];
+
+/*
+  public static $modules = [
+    'replication',
+  ];
+*/
+
+  public function setUp() {
+    parent::setUp();
+
+    $workspace_type_storage = \Drupal::entityManager()->getStorage('workspace_type');
+    $workspace_type = $workspace_type_storage->load('basic');
+
+    $is_workspace_moderatable = $workspace_type->getThirdPartySetting('workbench_moderation', 'enabled');
+    $this->assertFalse($is_workspace_moderatable);
+
+    // Make the "basic" Workspace moderatable.
+    $workspace_type->setThirdPartySetting('workbench_moderation', 'enabled', TRUE);
+    $workspace_type->setThirdPartySetting('workbench_moderation', 'allowed_moderation_states', ['draft', 'needs_review', 'published']);
+    $workspace_type->setThirdPartySetting('workbench_moderation', 'default_moderation_state', 'draft');
+    $workspace_type->save();
+
+    $is_workspace_moderatable = $workspace_type->getThirdPartySetting('workbench_moderation', 'enabled');
+    $this->assertTrue($is_workspace_moderatable);
+  }
+
+  /**
+   * Test replication settings with the InternalReplicator.
+   */
+  public function testReplicate() {
+    $permissions = [
+      'create_workspace',
+      'edit_own_workspace',
+      'view_own_workspace',
+      'create test content',
+      'access administration pages',
+      'administer taxonomy',
+      'administer menu',
+      'access content overview',
+      'administer content types',
+      'administer node display',
+      'administer node fields',
+      'administer node form display',
+    ];
+
+    $container = \Drupal::getContainer();
+    $workspace_manager = $container->get('workspace.manager');
+
+    $this->createNodeType('Test', 'test');
+
+    $this->setupWorkspaceSwitcherBlock();
+
+    $test_user = $this->drupalCreateUser($permissions);
+    $this->drupalLogin($test_user);
+
+    $session = $this->getSession();
+
+    // Validate Workspaces are moderatable.
+    $this->drupalGet('/admin/structure/workspace/add');
+    $this->assertEquals(200, $session->getStatusCode());
+    $page = $session->getPage();
+    $this->assertTrue($page->hasContent('Save and Create New Draft'), 'Workspaces are moderatable');
```

Comment:
This line fails. This is confusing given that https://github.com/timmillwood/drupal-workspace/pull/7/files#diff-54e0141e5c67fd897de6f5e7fb6d046dR63 passes!

Any thoughts on how this could be?